src/fsharp/PostInferenceChecks.fs

```diff
@@ -455,6 +455,27 @@ let CheckMultipleInterfaceInstantiations cenv interfaces m =
     | Some (typ1,typ2) -> 
          errorR(Error(FSComp.SR.chkMultipleGenericInterfaceInstantiations((NicePrint.minimalStringOfType cenv.denv typ1), (NicePrint.minimalStringOfType cenv.denv typ2)),m))
 
+// tries to extract the name of an expression
+let tryExtractNameOf args =
+    match args with
+    | [Expr.Val(r,_,_)] -> Some(r.CompiledName)
+    | [Expr.App(Expr.Val(r,_,_),_,_,Expr.Const(constant,_,_)::_,_)] -> 
+        if r.CompiledName.StartsWith("get_") && constant = Const.Unit then // TODO: We need a better way to find static property getters
```

Comment:
I think you want to use

        r.IsMember && match membInfo.MemberFlags.MemberKind with MemberKind.PropertyGet -> true | _ -> false

However note this is only true for F#-defined members.  Please test on IL-defined members. Unfortunately these come through as Expr.ILCall nodes.

It also looks like this won't work for properties that only have setter members.  That's unfortunate but is intrinsic to how this implementation is approaching things, where the argument to ``nameof`` gets checked by type checking and than transformed.  I believe we should reconsider the implementation of the feature and intercept nameof during type checking.  Instead of checking the argument we would require that it is a long identifier and do unqualified expression name resolution on that identifier to a symbol, and then just take the name of the symbol.  I believe that is more in the spirit of how the feature is intended to work.